[Ponharish] iP

Ponharish opened the pull request:
# MEEJU 🐈
**Your Personal assistant to track your tasks**

## Who is Meeju?
This application is named after my pet cat ***MEEJU***
![WhatsApp Image 2024-08-18 at 22 16 52](https://github.com/user-attachments/assets/9d010295-56ce-4aa9-ab35-f2a02a7b38ea)


## Advantages of using this Update

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

## Added features

1. Listing task
2. Adding different types of tasks such as event, todo, deadline
3. Deleting a task

## How to use this?

1. download it from [here](https://github.com/Ponharish/ip/releases/tag/v0.1).
2. download the zip file/ JAR file
3. add your tasks.
4. let it manage your tasks for you 😉

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Ponharish committed:
Rename Duke.java to Meeju.java
Add GoodByeMessage.java

Ponharish committed:
Add UserInteraction.java
Update Meeju.java, GoodByeMessage.java

Ponharish committed:
Add Task.java TaskCollection.java
Update UserInteraction.java Meeju.java

Ponharish committed:
Update Task.java TaskCollection.java UserInteraction.java

Ponharish committed:
Add Deadline.java Event.java Todo.java
Update Task.java TaskCollection.java UserInteraction.java

Ponharish committed:
Automated Text UI Testing

Ponharish committed:
Add MeejuException.java
Add code for exception handling

Ponharish committed:
Update TaskCollection.java UserInteraction.java
performed test again

Ponharish committed:
Update Storage.java

Add serialiser method to all task classes,
Rename TaskCollection.java to Storage.java,
implement File Handling

Ponharish committed:
Merge branch 'branch-Level-7'

Ponharish committed:
Update Deadline.java, Event.java, Storage.java

Supports Date and Time

Ponharish committed:
Merge branch 'branch-Level-8'

Ponharish committed:
Rename Storage.java to TaskList.java
Add Storage.java, Parser.java

Ponharish committed:
Add Package Meeju

Ponharish committed:
Merge branch 'add-gradle-support'

Ponharish committed:
Add JUnit Test

Add TaskTest.java, Parser.java, StorageStub.java
Rename Package Meeju to meeju

Ponharish committed:
Create Jar File

Ponharish committed:
Add JavaDocs

Ponharish committed:
Merge branch 'branch-A-JavaDoc'

Ponharish committed:
Update Storage.java, TaskList.java

Ponharish committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/meeju/Storage.java
#	src/main/java/meeju/TaskList.java

Ponharish committed:
Add findTask

Ponharish committed:
Merge branch 'branch-Level-9'

Ponharish committed:
Update TaskList.java

Ponharish committed:
Add checkStyle Files
Fix all style errors

Ponharish committed:
Merge branch 'A-CheckStyle'

Ponharish committed:
Add GUI upgrade layout

Ponharish committed:
Add GUI files

Ponharish committed:
Update TaskList.java, Meeju.Java, parser.java

Ponharish committed:
Update ParserTest.java
Fix Style Errors

Ponharish committed:
Merge branch 'Level-10'

yyanyee commented on src/main/java/meeju/MainWindow.java:
```diff
@@ -0,0 +1,65 @@
+                DialogBox.getMeejuDialog(response, meejuImage)
+        );
+        userInput.clear();
+        if (input.equals("bye")) {
```
`"bye"` can be extracted into a constant

yyanyee replied:
`"bye"` can be extracted to a constant

yyanyee commented on src/main/java/meeju/MainWindow.java:
```diff
@@ -0,0 +1,65 @@
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
```
documentation should clarify that `handleUserInput` terminates the application when the user says 'bye'

yyanyee commented on src/main/java/meeju/Parser.java:
a lot of repetition and code duplication with the if-else blocks. in particular, the commands have a similar structure where they check if the instruction starts with a keyword before calling a method from `TaskList`.

you can consider using `switch` instead.

yyanyee commented on src/main/java/meeju/Storage.java:
the class has good and concise documentation! 👍

yyanyee commented on src/main/java/meeju/Storage.java:
```diff
@@ -0,0 +1,104 @@
+            try {
+                this.scanner = new Scanner(this.file);
+            } catch (Exception e) {
+                System.out.println(e); //Ideally never reached here!
```
exception can be handled more gracefully by informing the user of the issue instead of just printing out the exception.

for example:
`System.out.println("Error reading file: " + e.getMessage());
`

yyanyee commented on src/main/java/meeju/Task.java:
```diff
@@ -0,0 +1,50 @@
+     * @param taskDescription The description of the task.
+     * @param isDone The status of the task.
+     */
+    public Task(String taskDescription, Boolean isDone) {
```
`boolean` should be used instead of its wrapper class

Sornsornah commented on src/main/java/meeju/GoodByeMessage.java:
```diff
@@ -0,0 +1,18 @@
+     *
+     */
+
+    public static String goodByeMessage() {
```
Should the method name be `getGoodByeMessage()` to make it sound more like an action?

Sornsornah commented on src/main/java/meeju/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
Would it be clearer if what is flipped was stated (e.g. `flipDialogBox()`)?

Sornsornah commented on src/main/java/meeju/Storage.java:
```diff
@@ -0,0 +1,104 @@
+            }
+            while (scanner.hasNext()) {
+                String currentLine = scanner.nextLine();
+                String[] currentLineParsed = currentLine.split("!-");
```
Should the variable name be `currentParsedLines` to follow the plural convention?

Sornsornah commented on src/main/java/meeju/Task.java:
```diff
@@ -0,0 +1,50 @@
+        this.isDone = isDone;
+    }
+
+
```
Is there an additional blank line here?

Sornsornah commented on src/main/java/meeju/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+    }
+
+
+    public int getNumberOfTask() {
```
Should the method name be changed to make it clearer (e.g. `getNumberOfTasks()` )?

Sornsornah commented on src/test/java/meeju/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Test
+    public void add_todo_task_Success() {
```
I noticed that you tried to follow the method naming convention of tests, i.e. `featureUnderTest_testScenario_expectedBehavior()`.
Could each part be clearer to explain the test?

- for `featureUnderTest`, could it be `addToDo`? Having an additional `_` might result in confusion when reading the test case
- for `expectedBehavior`, would it be possible to define the expected behaviour such as `taskAdded`?

Sornsornah reviewed (commented):
Overall, the code is very easy to read since comments were given to explain each design consideration! I made a few suggestions regarding naming conventions for your consideration.

Sornsornah reviewed (commented):
Overall, your code was very easy to read as comments were made that explained your design considerations. I have made some suggestions regarding some naming conventions.

Ponharish committed:
Update Parser.java

Ponharish committed:
Update Parser.java

Ponharish committed:
Add Assertions

Assertions added to Parser.java, Task.java, TaskList.java, Event.java, Deadline.java

Having assertions allows us to test our assumptions about certain segments of the code.
These assertions primarily focus on Null Pointers.

These conditions ensure that the assumptions are valid throughout the execution.

Ponharish committed:
Update Task.java

yyanyee reviewed (commented):
Overall good! Just some minor changes to ensure neater code

Ponharish committed:
Update Task.java, Storage.java

Ponharish committed:
update Todo.java

Ponharish committed:
update Deadline.java

Ponharish committed:
update Event.java

Ponharish committed:
update DialogBox.java

Ponharish committed:
update MainWindow.java

Ponharish committed:
update Meeju.java

Ponharish committed:
update Meeju.java, Parser.java

Ponharish committed:
update TaskList.java

Ponharish committed:
update Test Files

Ponharish committed:
Merge pull request #1 from Ponharish/A-Assertions

Merge branch 'A-Assertions'

Ponharish committed:
Merge branch 'master' into A-CodeQuality

# Conflicts:
#	data/meeju.txt
#	src/main/java/meeju/Deadline.java
#	src/main/java/meeju/Event.java
#	src/main/java/meeju/Parser.java

Ponharish committed:
remove GoodByeMessage.java

Ponharish committed:
Merge pull request #2 from Ponharish/A-CodeQuality

A-CodeQuality

Ponharish committed:
update Task.java

The class has been modified to an abstract class as there will be no generic Task created.

Task class is inherited by Deadline, Todo and Event classes

Ponharish committed:
add Reminders feature

Ponharish committed:
Merge tag 'B-Reminders'

Ponharish committed:
update user interface

Ponharish committed:
update runtest.bat

Ponharish committed:
remove DaDuke.png, DaUser.png

Ponharish committed:
update DialogBox.fxml

Ponharish committed:
Add ui.png

Ponharish committed:
update MeejuException.java

Ponharish committed:
update Task.java

Ponharish committed:
update TaskList.java

Ponharish committed:
update README.md

Ponharish committed:
Merge branch 'A-UserGuide'

Ponharish committed:
update Readme.md

Ponharish committed:
update Readme.me

Ponharish committed:
Revert "update Readme.me"

This reverts commit 8648c3743e3d08aebb129c9ca34b4954f8d115e8.

Ponharish committed:
update Readme.md

Ponharish committed:
update Readme.md

Ponharish committed:
remove ui.png

Ponharish committed:
add Ui.png

Ponharish committed:
Update README.md

Ponharish committed:
update ParserTest.java, TaskTest.java

Ponharish committed:
Merge branch 'A-MoreTesting'

Ponharish committed:
update Storage.java

Ponharish committed:
update ParserTest.java
